domain/application/service/application.go

```diff
@@ -1420,6 +1428,31 @@ func (s *Service) GetApplicationEndpointBindings(ctx context.Context, appID core
 	return bindings, errors.Capture(err)
 }
 
+// MergeApplicationEndpointBindings merge the provided bindings into the bindings
+// for the specified application.
+// The following errors may be returned:
+// - [applicationerrors.ApplicationNotFound] if the application does not exist
+func (s *Service) MergeApplicationEndpointBindings(ctx context.Context, appID coreapplication.ID, bindings map[string]network.SpaceName, force bool) error {
+	ctx, span := trace.Start(ctx, trace.NameFromFunc())
+	defer span.End()
+
+	if err := appID.Validate(); err != nil {
+		return errors.Errorf("validating application ID: %w", err)
+	}
+
+	// NOTE: This check is done here to avoid the force flag finding its way
```

Comment:
I don't think we need this comment. The situation you describe is vanishingly improbable - the application's charm would have to change between the calls to validate and merge.

Reply:
Yep fair enough